app/models/suggested_edit.rb

```diff
@@ -10,6 +10,12 @@ class SuggestedEdit < ApplicationRecord
   has_and_belongs_to_many :tags
   has_and_belongs_to_many :before_tags, class_name: 'Tag', join_table: 'suggested_edits_before_tags'
 
+  after_save :clear_pending_cache
```

Comment:
This should improve the efficiency (note I did not check this directly). Is there a better property to check whether it is valid?

```suggestion
  after_save :clear_pending_cache, if: :active
```

Reply:
I'm not sure that this works? Presumably this checks after the save, so if a suggested edit is accepted, then `:active` is false and so it doesn't clear the cache. At any rate, I don't think non-active suggested edits ever even get updated anyway.

Reply:
Ah, then I misunderstood, sorry.

Reply:
Cache should be cleared when a suggested edit changes `:active` from true to false (theoretically also false to true, but un-rejecting or un-accepting isn't a thing so this case doesn't matter)

Reply:
Then it should be an if statement for
`saved_change_to_attribute?(:active, from: true, to: false)`
That can either be in the method or as a proc to the if parameter

```suggestion
  after_save :clear_pending_cache, if: Proc.new { saved_change_to_attribute?(:active, from: true, to: false) }
```

(I wrote this on my phone, please double check for mistakes 🙈 )

Reply:
(See https://api.rubyonrails.org/classes/ActiveRecord/AttributeMethods/Dirty.html#method-i-saved_change_to_attribute-3F )

Reply:
This works, thanks 👍